neuralpredictors/layers/readouts.py

```diff
@@ -988,11 +1003,24 @@ def forward(self, x, sample=None, shift=None, out_idx=None):
         if shift is not None:
             grid = grid + shift[:, None, None, :]
 
-        y = F.grid_sample(x, grid, align_corners=self.align_corners)
-        y = (y.squeeze(-1) * feat).sum(1).view(N, outdims)
+        if not multiplex:
```

Comment:
```suggestion
       if multiplex:
            y = torch.einsum("ncwh,uco->nwho", x, feat)
        else:
            y = F.grid_sample(x, grid, align_corners=self.align_corners)
            y = (y.squeeze(-1) * feat).sum(1).view(N, outdims)
```

I think checking negatives and having the else clause makes it a bit more difficult to follow.